optimum/intel/openvino/modeling_decoder.py

```diff
@@ -237,6 +237,13 @@ def reshape(self, batch_size: int, sequence_length: int):
         logger.warning("Static shapes are not supported for causal language model.")
         return self
 
+    def compile(self):
+        if self.request is None:
+            logger.info("Compiling the model...")
+            cache_dir = Path(self.model_save_dir).joinpath("model_cache")
+            ov_config = {**self.ov_config, "CACHE_DIR": str(cache_dir)}
+            self.request = core.compile_model(self.model, self._device, ov_config).create_infer_request()
```

Comment:
could be replaced by : 
```suggestion
            super().compile()
            self.request = self.request.create_infer_request()
```